### PR TITLE
Add amending-commits.md

### PR DESCRIPTION
**Pull request by chriscool**
This could help people improve their commits after tests or code reviews complain about them.

### CODE REVIEW

File: docs/amending-commits.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+commits on the current branch. Just run:
+
+```
+$ git rebase -i HEAD~5
+```
+
+It will open an instruction sheet in your editor where you can specify
+how commits should be rebased. There you just need to replace the
+"pick" instruction at the beginning of some lines with "r" (or
+"reword").
```
Comment (jbenet) [inline docs/amending-commits.md]: maybe show an example, or something:

---

"reword"). For example:

```
r 06f6e99 merkledag spec
pick 02b20bf records spec
```

```

File: docs/amending-commits.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+for the missing trailers if any to be added.
+
+* Improving the commit message of any commit
+
+It's also easy to improve the commit message of let's say the last 5
+commits on the current branch. Just run:
+
+```
+$ git rebase -i HEAD~5
+```
```
Comment (jbenet) [inline docs/amending-commits.md]: may want to have a section on rebasing, and show how to rebase against new origin/master. (that's a very common case)

**Comment by jbenet** [comment]
@chriscool this is very good +1. couple of comments, otherwise LGTM

**Comment by wking** [comment]
Similar external-to-IPFS docs that we may want to consider pointing to
instead:

* The maintained-in-git.git guide [1].
* The embedded-in-git-scm.com book [2].

[1]: https://www.kernel.org/pub/software/scm/git/docs/user-manual.html#cleaning-up-history
[2]: http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by jbenet** [comment]
thanks @chriscool !

**Pull request merged**